[Fix] Update links to Music Blocks Guide

farhan-momin opened the pull request:
This PR addresses-

- Update Link to MB guide in Take a Tour Section.

- Update link of MB guide in `js/SaveInterface.js`. Now it will add correct link for guide in saved `.html` MB project files. Made changes to corresponding test file for the same. 

- Update MB guide link for projects in `/examples`.

Ubayed-Bin-Sufian commented:
The guide link is optimized which is good start. Clicking the link shows error which is not supposed to happen. I had to manually  delete the guide.html.
https://github.com/user-attachments/assets/6aee8e62-9496-4b40-b75d-65cce258a120

farhan-momin commented:
@Ubayed-Bin-Sufian thankyou for reviewing the pr! I checked the same MB guide link again it does open the guide link at http://localhost:3000/guide/.

Can you try clearing the cookies and site data and then click the link? Because i was able to reproduce the issue you described, when i opened MB on localhost without clearing it.

Ubayed-Bin-Sufian commented:
@walterbender Please review.

walterbender commented:
I don't understand the approach here. Why not simply correct the constant defined in turtledefs.js? Also, how are the changes to examples relevant?

farhan-momin commented:
@walterbender I changed the guide links in examples because they didnt work.

farhan-momin commented on js/turtledefs.js:
Here the link didnt work after https://github.com/sugarlabs/musicblocks/commit/3e1fdee7b8c74d14a6144d494ee711b28cb64b09 so I updated it.

farhan-momin commented on js/turtledefs.js:
```diff
@@ -68,7 +68,7 @@ if (_THIS_IS_TURTLE_BLOCKS_) {
 
     if (GUIDEURL === "guide url" || GUIDEURL === "") {
         GUIDEURL =
-            "../guide/guide.html";
```
Here the link didnt work after https://github.com/sugarlabs/musicblocks/commit/3e1fdee7b8c74d14a6144d494ee711b28cb64b09 so I updated it.

walterbender commented:
I still don't understand why you don't update the definition of GUIDEURL in turtledefs instead.

Ubayed-Bin-Sufian commented on js/__tests__/SaveInterface.test.js:
```diff
@@ -36,7 +36,6 @@ global.jQuery.noConflict = jest.fn(() => global.jQuery);
 global._ = jest.fn((str) => str);
 global._THIS_IS_TURTLE_BLOCKS_ = true;
 global.TITLESTRING = "Music Blocks";
-global.GUIDEURL = "../guide/guide.html";
```
This a global variable. I think edit it instead of deleting it.

farhan-momin commented:
> I still don't understand why you don't update the definition of GUIDEURL in turtledefs instead.

@walterbender Actually I have just updated the definition of GUIDEURL in turtledefs, and it is accessible globally. I tested it by successfully logging the GUIDEURL in other .js files which logs `/guide`.

I removed the global.GUIDEURL in [js/__tests__/SaveInterface.test.js](https://github.com/sugarlabs/musicblocks/pull/4613/files/e011780b3a7e9034275fe54585ea5dbee3575cf7#diff-1b1cab1e2be92a2ef466fc9a38678c1affe4fd574eb09b5c3b9889a90d65e48c) because it was not used in any file after the changes. But i think i should roll it back because it is a global variable.

Please let me know how should I proceed

walterbender commented:
It should be declared as a global.

farhan-momin commented:
> It should be declared as a global.

To address this, should I add `GUIDEURL` in the` /* exported */` list at [turtledefs.js](https://github.com/sugarlabs/musicblocks/blob/master/js/turtledefs.js)? Please correct me if I am wrong.

@walterbender @Ubayed-Bin-Sufian

walterbender commented:
Yes. It should be exported where it is defined and defined as global where it is used.

farhan-momin commented:
I've added a new [commit](https://github.com/sugarlabs/musicblocks/pull/4613/commits/dd28530a6d81f14707f656dda65308c7f077c61f) to include `GUIDEURL` in the exported section of `turtledefs.js`. 
@walterbender please review

walterbender commented on js/SaveInterface.js:
```diff
@@ -12,7 +12,7 @@
 /*
    globals
 
-   _, TITLESTRING, GUIDEURL, jQuery, docById, docByClass, doSVG,
```
Why cannot we use the global here?

farhan-momin commented on js/SaveInterface.js:
```diff
@@ -70,9 +70,7 @@ class SaveInterface {
-            ' <a href="' +
-            GUIDEURL +
-            '" target="_blank">' +
+            ' <a href="https://musicblocks.sugarlabs.org/guide" target="_blank">' +
```
> Why cannot we use the global here?

@walterbender The reason is previously the guide link in saved project `.html` files were relative links, so if the project files were saved outside the MB project folder, they wouldn't work. The same happens for project files downloaded from MB website.

 So i replaced that with MB website guide link. This will allow MB guide access to project files in any location. 
 
 Still we can use `GUIDEURL` with the following approach-
```
<a href="https://musicblocks.sugarlabs.org'+
            GUIDEURL +
            '" target="_blank">'
```

Please Let me know if this approach would be preferred instead of the original

walterbender commented:
OK. Makes sense. Is the link wrong in all of the example files? (I noticed you updated just a few of them.)

farhan-momin commented:
@walterbender The example files in which i updated the guide link had `"../guide/guide.html"` as the guide link.

Rest of the example files have `"https://github.com/sugarlabs/musicblocks/tree/master/guide/README.md"` as the guide link and they work. So i did not change those.

But now it clicked for me that the guide links in these examples point to the master branch of MB on github, which is the development branch. So it can guide about things which are unreleased on MB website.

Should i change those to `"https://musicblocks.sugarlabs.org/guide"` too?

walterbender commented:
Prob. we should always point to https://musicblocks.sugarlabs.org/guide/guide.html

farhan-momin commented:
> Prob. we should always point to https://musicblocks.sugarlabs.org/guide/guide.html

@walterbender Ok. Since the guide page on https://musicblocks.sugarlabs.org/guide/guide.html will be on https://musicblocks.sugarlabs.org/guide, i will update all example files to point to `https://musicblocks.sugarlabs.org/guide`

farhan-momin commented:
> @walterbender Ok. Since the guide page on https://musicblocks.sugarlabs.org/guide/guide.html will be on https://musicblocks.sugarlabs.org/guide, i will update all example files to point to `https://musicblocks.sugarlabs.org/guide`

@walterbender I've added remaining project files with updated guide links. please review

walterbender commented:
Isn't it supposed to be https://musicblocks.sugarlabs.org/guide/guide.html ? Unless we want to mv guide.html to index.html

farhan-momin commented:
> Isn't it supposed to be https://musicblocks.sugarlabs.org/guide/guide.html ? Unless we want to mv guide.html to index.html

Since the guide link is updated in https://github.com/sugarlabs/musicblocks/commit/3e1fdee7b8c74d14a6144d494ee711b28cb64b09 the guide page will be on https://musicblocks.sugarlabs.org/guide instead.

I have tested it by visiting https://sugarlabs.github.io/musicblocks/guide
